plugins/item_tasks/server/rest.py

```diff
@@ -130,8 +131,25 @@ def _transformInputs(self, inputs, token):
                 if rtype not in {'file', 'item', 'folder'}:
                     raise ValidationException('Invalid input resource_type: %s.' % rtype)
 
-                resource = self.model(rtype).load(
-                    v['id'], level=AccessType.READ, user=self.getCurrentUser(), exc=True)
+                try:
+                    resource = self.model(rtype).load(
+                        v['id'], level=AccessType.READ, user=self.getCurrentUser(), exc=True)
+                except ValidationException:
+                    # if we asked for a file, we may have been given an item,
+                    # which case get the first file within it.  If the item's
+                    # metadata includes a dictionary that includes fileId, use
+                    # that file id.
+                    if rtype != 'file':
+                        raise
+                    item = Item().load(
+                        v['id'], level=AccessType.READ, user=self.getCurrentUser(), exc=True)
+                    fileIds = [v['fileId'] for k, v in six.iteritems(item)
+                               if isinstance(v, dict) if v.get('fileId')]
```

Comment:
I assume this is to get the large image `fileId` if it exists.  Is there any other use case for this behavior or would referencing `item.get('largerImage')` be sufficient?

Reply:
`item.get('largeImage')` would work, but it feels like that should be in the large_image plugin (which would then need to check if the item_tasks plugin exists).

Reply:
I agree it would be best to have an extensibility hook here and add the code in large_image, but that would clearly require more work.  Failing that, I would rather just encode the large_image behavior directly here for simplicity.